keel/api/v1/tasks/views.py

```diff
@@ -179,23 +227,33 @@ def taskDetails(self, request, format = 'json', **kwargs):
         user_id = user.id
         task_id = kwargs.get("task_id")
         try:
-            task = Task.objects.prefetch_related("tasks_comment","tasks_docs").get(task_id = task_id, user_id = user_id)
-            # task = Task.objects.get(task_id = task_id, user_id = user_id)
-            task_data = TaskSerializer(task)
-            resp_data = task_data.data
-            response['data'] = resp_data
-
-        except Task.DoesNotExist:
-            log_error("ERROR","GetTaskDetails:fetch", str(user_id), err = "Task DoesNotExist", task_id = task_id)
-            response['status'] = 1
-            response['message'] = 'Task Id is invalid'
-            resp_status = HTTP_STATUS.HTTP_400_BAD_REQUEST
+            task_serializer = TaskIDCheckSerializer(data = {"task_id": task_id})
+            task_serializer.is_valid(raise_exception = True)
+            task_obj = task_serializer.validated_data
 
-        except Exception as e:
+        except ValidationError as e:
             log_error("ERROR","GetTaskDetails:fetch", str(user_id), err = str(e), msg = "Unknown exception")
             response['status'] = 1
             response['message'] = GENERIC_ERROR
             resp_status = HTTP_STATUS.HTTP_500_INTERNAL_SERVER_ERROR
+            return Response(response, resp_status)
+
+
+        case_id = task_obj.case_id
+
+        ## Validate User with Task.Case, so that only Case's Customer/RCIC shud be fetching task details
+        try:
+            case_serializer = CaseIDSerializer(data = {"case_id": case_id, "user_id": user_id})
+            case_serializer.is_valid(raise_exception=True)
+            case_obj = case_serializer.validated_data
+        except ValidationError as e:
```

Comment:
here also...Serializer's Exception Would be overriden by custom exception handling!

Reply:
Updated, using the default response exception handler.